tango/executor.py

```diff
@@ -88,14 +94,21 @@ def execute_step_graph(
                 try:
                     self.execute_step(step)
                     successful.add(step.name)
-                except Exception:
+                except Exception as exc:
                     failed.add(step.name)
-                    error_tracebacks.append(traceback.format_exc())
-
-        for stacktrace in error_tracebacks:
-            logger.error(stacktrace)
+                    log_exception(exc, logger)
 
         return ExecutorOutput(successful=successful, failed=failed, not_run=not_run)
 
+    def execute_sub_graph_for_step(
+        self, step_graph: StepGraph, step_name: str, run_name: Optional[str] = None
+    ) -> None:
+        """
+        Execute the sub-graph associated with a particular step in a
+        :class:`~tango.step_graph.StepGraph`.
+        """
+        step = step_graph[step_name]
```

Comment:
We should add more info in the docstring here explaining why we need this separate method -- it'll be easy to lose/forget the information discussed in this PR.